pyani_plus/workflows/snakemake_sourmash.smk

```diff
@@ -42,32 +42,31 @@ rule sketch:
         kmersize=config["kmersize"],
     input:
         genomeA=get_genomeA,
-        genomeB=get_genomeB,
     output:
-        "{outdir}/{genomeA}_vs_{genomeB}.sig",
+        "{outdir}/{genomeA}.sig",
     shell:
-        "sourmash sketch dna -p 'k={params.kmersize},{params.extra}' {input.genomeB} {input.genomeA} -o {output} > {output}.log 2>&1"
+        "sourmash sketch dna -p 'k={params.kmersize},{params.extra}' {input} -o {output} > {output}.log 2>&1"
 
 
 rule compare:
     params:
         db=config["db"],
+        run_id=config["run_id"],
         outdir=config["outdir"],
         mode=config["mode"],
         kmersize=config["kmersize"],
         extra=config["extra"],  # This will consist of either `scaled=X` or `num=X`.
     input:
-        "{outdir}/{genomeA}_vs_{genomeB}.sig",
-        genomeA=get_genomeA,
-        genomeB=get_genomeB,
+        expand("{{outdir}}/{genome}.sig", genome=sorted(indir_files)),
```

Comment:
This gets used in the command line as a list of that many filenames… is there a buffer size upper limit that we might encounter here?

Reply:
Maybe write these filenames to a file and use redirection? (does this still potentially hit the limit)?